src/qmapshack/poi/CPoiPOI.h

```diff
@@ -22,18 +22,49 @@
 #include "poi/IPoi.h"
 
 #include <QMutex>
+#include <QTimer>
 
 class CPoiPOI : public IPoi
 {
 public:
     CPoiPOI(const QString& filename, CPoiDraw *parent);
     virtual ~CPoiPOI() = default;
 
+    void addTreeWidgetItems(QTreeWidget *widget) override;
+    void checkedStateChanged(QTreeWidgetItem*item, int columnChanged) override;
+    // category, minLon multiplied by 10, minLat multiplied by 10. POIs are loaded in squares of degrees (should be fine enough to not hang the system)
+    void loadPOIsFromFile(const QString &category, int minLonM10, int minLatM10);
+
     void draw(IDrawContext::buffer_t& buf) override;
 
+    static void init()
+    {
+        tagMap = initTagMap();
+    }
 private:
+    struct poi_t
+    {
+        QStringList data;
+        QPointF coordinates;
+    };
+    enum SqlColumn_e
+    {
+        eSqlColumnMaxLat,
+        eSqlColumnMaxLon,
+        eSqlColumnMinLat,
+        eSqlColumnMinLon,
+        eSqlColumnData
+    };
+
     QMutex mutex;
     QString filename;
+    QTimer loadTimer;
+    QMap<QString, bool> categoryActivated;
```

Comment:
You use this map to store the check state converting the check state into a bool and vice versa. Wouldn't it be much more straight forward to use the check state?